ferminet/networks.py

```diff
@@ -115,7 +115,7 @@ def init_fermi_net_params(
       'single': [{} for i in range(len(hidden_dims))],
       'double': [{} for i in range(len_double)],
       'orbital': [],
-      'envelope': [{} for spin in spins]
+      'envelope': [{} for spin in spins if spin > 0]
```

Comment:
I wonder if, rather than modifying every use of `spins` in the initialisation, we should just filter out `spin = 0` from `spins` either at the start of this function or in `make_fermi_net`.

I think it's probably best not to modify spins in `make_fermi_net` (as knowing which spin channels contain a non-zero number of electrons is useful).

How about `active_spin_channels = [spin for spin in spins if spin > 0]` and then use `active_spin_channels` instead in `init_fermi_net_params`?